common/lib/day.js

```diff
@@ -1,13 +1,18 @@
 const moment = require('moment')
 
 class Day {
-  constructor (yearOrMoment, month, day) {
+  constructor (yearDateStringOrMoment, month, day) {
     if (arguments.length === 0) {
       this.moment = moment.utc()
-    } else if (moment.isMoment(yearOrMoment)) {
-      this.moment = yearOrMoment
+    } else if (
+      typeof yearDateStringOrMoment === 'string' &&
+      yearDateStringOrMoment.match(/([0-9]{4}(-[0-9]{1,2}){1,2})\b/)
```

Comment:
This isn't quite the right regex, it will allow some very invalid stuff through. We should have a quick chat about this.

Reply:
Fixed :)